src/ros_filter.cpp

```diff
@@ -557,6 +590,15 @@ namespace RobotLocalization
     // Determine if we're in 2D mode
     nhLocal_.param("two_d_mode", twoDMode_, false);
 
+    // Smoothing window size.
+    //
+    // Internally stored in ms because checking double values to be exactly zero is unreliable.
```

Comment:
Another way around this that might be to store two variables: useFilterHistory_ (bool) and filterHistoryInterval_ (double). Let the parameter be a double and just do something like `useFilterHistory_ = ::fabs(filterHistoryInterval_) > 1e-9`. You should also check for invalid interval values (i.e., negative ones), and warn the user.